statement_test.go

```diff
@@ -497,3 +498,91 @@ func TestGetQueryID(t *testing.T) {
 		t.Fatalf("failed to prepare statement. err: %v", err)
 	}
 }
+
+func TestAsyncMode(t *testing.T) {
```

Comment:
I would suggest to add test cases in which several queries within the same session or in different sessions will be filed asynchronously, and some results of late queries could return earlier.